src/js/index.js

```diff
@@ -0,0 +1,138 @@
+        
+        btns.forEach(btn => {
+            btn.onclick = function(){
+            cartObj.addItem(btn.parentElement.parentElement.id || btn.parentElement.parentElement.parentElement.id);
+            
+            
+          }
+        });
+      }
+
+      function setCategoryListeners(){
+        var categories = document.querySelectorAll(".category");
+        categories.forEach(category => {
+          category.onclick = function(event){
+            var categoryId= event.target.id || (event.target.id).split("-")[1];
+            plpPage.RenderFilteredItems(categoryId);
+            setBuyBtnEventListner();
+        }
+      });
+
+      }
+
+      function loadBanners(){
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.open( "GET", "api/banners/index.get.json", false ); // false for synchronous request
+        xmlHttp.send( null );
+        
+        var bannersJSON = JSON.parse(xmlHttp.responseText);
+        var bannersJSON = bannersJSON.sort(function(a, b) {return a.order - b.order});
+        var data = "{ \"banners\":"+JSON.stringify(bannersJSON)+"}";
+        //console.log(data);
+        bannersJSON = JSON.parse(data);
+        var container = document.querySelector('.carousel-inner');
+        container.innerHTML = bannerTemplate(bannersJSON);
+        container.firstElementChild.classList.add("active");
+
+        // load categories
+
+        var xmlHttp = new XMLHttpRequest();
+        xmlHttp.open( "GET", "api/categories/index.get.json", false ); // false for synchronous request
+        xmlHttp.send( null );
+        
+        var categoriesJSON = JSON.parse(xmlHttp.responseText);
+        categoriesJSON.forEach(function(el){el.direction=1});
+        var categoriesJSON = categoriesJSON.sort(function(a, b) {return a.order - b.order});
+        var data = "{ \"categories\":"+JSON.stringify(categoriesJSON)+"}";
+        //console.log(data);
+        categoriesJSON = JSON.parse(data);
+        var container = document.getElementById('categories-banners');
+        container.innerHTML = categoriesBannerTemplate(categoriesJSON);
+        
+
+
+      }
+
+  return {
+    init : function(){
+      initCart();  
+    },
+    loadBanners: function(){
+      loadBanners();
+    },
+
+    loadPlp : function(){
+      console.log("loading products");
+      
+      plpPage.RenderProducts();
+      setBuyBtnEventListner();
+      setCategoryListeners();  
+
+    }
+  }
+})();
+
+controller.init();
+
+try{
+  controller.loadBanners();
+}catch(err){
+}
+try{
+  controller.loadPlp();
+}catch(err){
+}
+
+
+
+ // Get the modal
+      var modal = document.getElementById('myModal');
+
+      // Get the button that opens the modal
+      var btn = document.getElementById("cart");
+      
+      // Get the <span> element that closes the modal
+      var span = document.getElementsByClassName("close")[0];
+      
+      // When the user clicks on the button, open the modal 
+      btn.onclick = function() {
```

Comment:
These are not right ways to attach events. Why these events are not attached inside controller.init();